index.bs

```diff
@@ -1440,3 +1467,281 @@ The [=remote end steps=] with |command parameters| are:
     1. Return the result of [=updating the event map=] with [=current session=],
        |list of event names|, |list of contexts| and enabled false.
 </div>
+
+
+## The browsingContext Module ## {#module-browsingContext}
+
+The <dfn export for=modules>browsingContext</dfn> module contains commands and
+events relating to browsing contexts.
+
+A [=/session=] has a <dfn>context discard</dfn> flag which is initially
+unset.
+
+### Definition ### {#module-browsingContext-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+BrowsingContextCommand = (BrowsingContextGetTreeCommand)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+BrowsingContextResult = (BrowsingContextGetTreeResult)
+
+BrowsingContextEvent = (
+    BrowsingContextCreatedEvent //
+    BrowsingContextDestroyedEvent
+)
+
+</pre>
+
+### Types ### {#module-browsingcontext-types}
+
+#### The browsingContext.BrowsingContext Type #### {#types-browsingcontext}
+
+[=remote end definition=] and [=local end definition=]
+
+<pre class="cddl remote-cddl local-cddl">
+BrowsingContext = text;
+</pre>
+
+Each [=/browsing context=] has an associated <dfn export>browsing context
+id</dfn>, which is a string uniquely identifying that browsing context. This is
+implicitly set when the context is created. For browsing contexts with an
+associated WebDriver [=window handle=] the [=/browsing context id=] must be the
+same as the [=window handle=].
+
+<div algorithm>
+To <dfn>get a browsing context</dfn> given |context id|:
+
+ 1. If |context id| is null, return [=success=] with data null.
+
+ 1. If there is no browsing context with [=browsing context id=] |context id| return
+    [=error=] with [=error code=] [=no such frame=]
+
+ 1. Let |context| be the browsing context with id |context id|.
+
+ 1. Return [=success=] with data |context|
+
+</div>
+
+#### The browsingContext.BrowsingContextInfo Type #### {#types-browsingcontextinfo}
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+BrowsingContextInfoList = [* BrowsingContextInfo]
+
+BrowsingContextInfo = {
+  context: BrowsingContext,
+  ?parent: BrowsingContext / null,
+  url: text,
+  children: BrowsingContextInfoList / null
```

Comment:
```suggestion
  children: BrowsingContextInfoList
```

Being able to represent "no children" as both null and an empty list seems unnecessary.

Reply:
I think they mean slightly different things. `null` is used for "there may be additional contexts but we didn't recurse into them" and the empty list is used for "there are no child contexts". That's a subtle difference but it does seem useful to communicate it somehow.

Reply:
I see https://pr-preview.s3.amazonaws.com/w3c/webdriver-bidi/pull/62.html#get-the-descendent-browsing-contexts returns null if *max depth* is reached. Similar to remote value serialization. OK then :)